src/components/network/network-panel.js

```diff
@@ -137,6 +141,14 @@ genotet.NetworkPanel.prototype.setEdgeInfo_ = function(edge, container) {
     .text(edge.target.label);
   container.children('#weight').children('span')
     .text(edge.weight);
+  container.children('#rm-edge').children('button').click(function() {
+    this.signal('update', {
+      type: 'delete-edge',
+      source: edge.source.label,
```

Comment:
Looks like you do need the source and target ids. So I recommend passing in either:
- the entire edge object (natively in JS as a reference)
- the source and target id strings

If you pass the edge id, it would be sth like "A,B" and you still have the parse the string to get A and B.